src/parser.c

```diff
@@ -194,21 +194,24 @@ print_tldrpage(char const *input, char const *poverride)
         }
     }
 
-    construct_url(url, URLBUFSIZ, input, platform);
+    if (!getenv(PREVENT_UPDATE_ENV_VARIABLE)) {
+        construct_url(url, URLBUFSIZ, input, platform);
 
-    /* make clang's static analyzer happy */
-    output = NULL;
-    download_content(url, &output, 0);
-    if (output == NULL) {
-        construct_url(url, URLBUFSIZ, input, "common");
+        /* make clang's static analyzer happy */
+        output = NULL;
         download_content(url, &output, 0);
-        if (output == NULL)
-            return 1;
-    }
+        if (output == NULL) {
+            construct_url(url, URLBUFSIZ, input, "common");
+            download_content(url, &output, 0);
+            if (output == NULL)
+                return 1;
+        }
 
-    parse_tldrpage(output);
+        parse_tldrpage(output);
 
-    free(output);
+        free(output);
```

Comment:
Due to the new flow, this should have `return 0;` after it, and line 215 would have `return 1;`. Welcome to correct me if you think I'm wrong.

If auto updates are enabled, and we find new pages, we should return 0 because everything was successful.

However, if auto-updates are disabled or after updating the page was not found, we should return 1 to indicate we failed to get a page. (With your PR as-is, we return a success status but offer no output to users when the page doesn't exist.)

It may also be worth adding a print in tldr.c with how the user can check for updates, for example:

```
This page doesn't exist yet!
Submit new pages here: https://github.com/tldr-pages/tldr
Automatic updates are disabled, you could try updating the local database manually with: tldr --update
```
> Note in this case I'd expect the 3rd message to appear in red, and only if `TLDR_AUTO_UPDATE_DISABLED` it set.

Reference: https://github.com/tldr-pages/tldr-c-client/blob/main/src/tldr.c#L178

I'm partial to the output I proposed, so feedback from anyone is welcome.

Reply:
Agreed on the return value. I'd propose a slight change in the wording of the additional error message:

```
$ tldr non_existing_command       
This page doesn't exist yet!
Submit new pages here: https://github.com/tldr-pages/tldr
Checking the online database was skipped because automatic updates are disabled.
You could try updating the local database manually with: tldr --update
```

If any of those lines should be printed in color, I'd argue it should be the first one, since it's the one actually saying what the error was. The last two are a hint at best but since the env variable has to be set manually by the user, I don't think this should be considered an error.